src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,212 @@
+#:include "common.fypp"
+module stdlib_stats_distribution_PRNG
+    use stdlib_kinds, only: int8, int16, int32, int64
+    implicit none
+    private
+    integer, parameter :: MAX_INT_BIT_SIZE = bit_size(1_int64)
+    integer(int64), save :: st(4), si = 614872703977525537_int64
+    logical, save :: seed_initialized = .false.
+
+    public :: random_seed
+    public :: dist_rand
+    public :: jump
+    public :: long_jump
+
+
+    interface dist_rand
+    !! Version experimental
+    !!
+    !! Generation of random integers with different kinds
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure dist_rand_${t1[0]}$${k1}$
+        #:endfor
+    end interface dist_rand
+
+    interface random_seed
+    !! Version experimental
+    !!
+    !! Set seed value for random number generator
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+    !!
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure random_distribution_seed_${t1[0]}$${k1}$
+        #:endfor
+    end interface random_seed
+
+
+    contains
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    function dist_rand_${t1[0]}$${k1}$(n) result(res)
```

Comment:
Could this function also return an array of PRN?

Reply:
Sure, it can be done. The issue here is Probability Distribution and Statistical Functions, not random number generator. As I stated in comments in the initial round of PR, Fortran's standard random_number function may generate different random number based on the platform and algorithm used by the OS. Further, dist_rand function is meant to be used internally in other distributions modules, not for end user.
The random number generator provided in this module might not be the best one out there, the appearance of this module in Probability Distribution and Statistical Functions is to provide convenience for test purpose. There has already been an issue on Pseudorandom  Number Generator (#135).